models/model.js

```diff
@@ -28,3 +28,18 @@ exports.getArticleById = (id) => {
       }
     });
 };
+
+exports.generateArticlesArray = () => {
+  return db.query(`SELECT * FROM articles;`).then(({ rows }) => {
+    return Promise.all(
+      rows.map((article) => {
+        return db.query(`SELECT * FROM comments WHERE article_id = $1`, [
+          article.article_id,
+        ]).then(({ rows }) => {
+          article.comment_count = rows.length;
+          return article;
+        });
+      })
+    );
+  });
+};
```

Comment:
This does definitely work which is great, but its not very efficient, we should try to avoid multiple queries as much as possible, and here you have a query for every article which is definitely going to take a little while to do. Instead this query can be done in one using a join and an aggregate function to achieve the expected outcome.